CODE_OWNERSHIP.md

```diff
@@ -0,0 +1,55 @@
+# Code Ownership and AMP
+
+This document describes the code ownership model for the AMP open source
+project, and in particular the [AMP HTML GitHub project](https://github.com/ampproject/amphtml).
+
+The goals of enforcing AMP code ownership are as follows:
+
+* Provide the flexibility to specify a different set of owners for individual
+  directories / files in the repository
+* Allow for the recursive inheritance of ownership for files / directories based
+  on the ownership of parent directory(ies)
+* Prevent incoming pull requests from being merged until they are approved by a
+  sufficient number of owners so as to cover all files / directories in the
```

Comment:
do we need to support this? github reviewers system currently allows merging once there are no comment reviews and at least 1 reviewer approved.

I was trying to avoid having to create a separate PR status but it seems like it would be necessary.

Reply:
Suppose a PR changes files in `src/foo/`, and `src/bar/`, where both directories have a different set of owners. Will it be possible to merge the PR if an owner of `src/foo/` approves it, but not `src/bar/`?

Reply:
currently yes its possible (because of how github reviewers work), this was a feature we wanted o support and would need the custom status check.

i was hoping to rely more heavily on github reviewers API thats all. 🤷‍♂️

Reply:
This is a key requirement of Chromium's [code ownership model](https://chromium.googlesource.com/chromium/src/+/master/docs/code_reviews.md#owners-files), whose first line reads:

> In various directories there are files named OWNERS that list the email addresses of people qualified to review changes in that directory. You must get a positive review from an owner of each directory your change touches.

If our intent is to fully follow this model, then I believe we'd need to be more strict than GitHub's reviewers API.

Do you think we can work around this by enabling the "Require review from Code Owners" setting, as described in item 8 [here](https://help.github.com/articles/enabling-required-reviews-for-pull-requests/)?

Reply:
@tobie Thanks for refreshing my memory :)
@erwinmombay, let's chat offline about making the bot block PRs until they are approved by an owner from each directory being modified. Meanwhile, any objections to leaving this line as is?

Reply:
I had written a few thoughts down when we last talked about this. [Here](https://docs.google.com/document/d/17dJUB_2OINX89YZ9y4xNqtO6zUAwtPEhIA0qpqc4DLc/edit) is a cleaned-up version.

Reply:
@rsimha no objections. just wanted to double check!